nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -6846,18 +6846,23 @@ public FlowAnalysisResultEntity getFlowAnalysisResult(String processGroupId) {
     }
 
     public Stream<RuleViolation> getRuleViolationStream(String processGroupId) {
-        ProcessGroup processGroup = processGroupDAO.getProcessGroup(processGroupId);
```

Comment:
I have a question about the original implementation. Would it have allowed a user to view violations in a child process group of the current process group regardless of the level of permission the user had on that process group?

I see no evidence that it is making this check. Should it be?